src/usn.rs

```diff
@@ -0,0 +1,331 @@
+                }
+            }
+        );
+
+        records_per_chunk.flatten()
+    }
+}
+
+pub struct IterFileChunks<'c, T: ReadSeek> {
+    parser: &'c mut UsnParser<T>,
+    // The chunk size is larger than the parse size to ensure complete end record
+    chunk_size: usize,
+    // The parse size is smaller than the chunk size to ensure there is overlap for complete end record
+    search_size: usize,
+    // This is the relative offset of this chunk
+    chunk_start_offset: u64,
+}
+
+impl <'c, T: ReadSeek> Iterator for IterFileChunks <'c, T> {
+    type Item = DataChunk;
+
+    fn next(&mut self) -> Option<<Self as Iterator>::Item> {
+        while self.chunk_start_offset < self.parser._size {
+            // Create buffer for our data chunk
+            let mut buffer = vec![0u8; self.chunk_size];
+
+            // Get the current offset
+            let current_offset = self.chunk_start_offset;
+            
+            // Seek to where we start our chunk
+            match self.parser._inner_handle.seek(
+                SeekFrom::Start(current_offset)
+            ) {
+                Ok(_) => {},
+                Err(error) => {
+                    error!("{}", error);
+                    break;
+                }
+            }
+
+            // Read into buffer
+            let _bytes_read = match self.parser._inner_handle.read(
+                buffer.as_mut_slice()
+            ){
+                Ok(bytes_read) => bytes_read,
+                Err(error) => {
+                    error!("{}", error);
+                    return None
+                }
+            };
+
+            // Set the next chunk's offset
+            // Increment by search size and not chunk size
+            self.chunk_start_offset += self.search_size as u64;
+
+            // Return data chunk
+            return Some(
+                DataChunk{
+                    offset: current_offset,
+                    search_size: self.search_size,
+                    data: buffer
+                }
+            );
+        }
+        
+        None
+    }
+}
+
+#[derive(Debug)]
+pub struct DataChunk {
+    offset: u64,
+    search_size: usize,
+    data: Vec<u8>
+}
+
+impl DataChunk {
+    pub fn get_records(self) -> Vec<UsnEntry> {
+        trace!("Getting records for ChunkData at offset: {}", self.offset);
+
+        let record_iterator = IterRecords::new(
+            self.data, 
+            self.offset,
+            self.search_size
+        );
+
+        let records: Vec<UsnEntry> = record_iterator.collect();
+
+        return records;
+    }
+}
+
+pub struct IterRecords {
+    block: Vec<u8>,
+    start_offset: u64,
+    match_offsets: Vec<u64>,
+}
+
+impl IterRecords {
+    pub fn new(block: Vec<u8>, start_offset: u64, search_size: usize) -> IterRecords {
+        lazy_static! {
+            static ref RE_USN: bytes::Regex = bytes::Regex::new(
+                "(?-u)..\x00\x00\x02\x00\x00\x00"
+            ).unwrap();
```

Comment:
Maybe https://github.com/shepmaster/jetscii could make this go even faster :)

Reply:
And if you keep regex, it's would have been nicer to use `.expect('This is a valid regex')` to let the reader know that this cannot possibly fail.

Reply:
Going to leave Regex in for now because I would like at some point to account for version 3 and 4 which would change regex to `"(?-u)..\x00\x00(\x02|\x03|\x04)\x00\x00\x00"`. I did remove unwrap in favor of expect though.